Mathlib/Topology/TietzeExtension.lean

```diff
@@ -34,6 +40,90 @@ topological space, then specialize them to the case `X = s : Set Y`, `e = (↑)`
 Tietze extension theorem, Urysohn's lemma, normal topological space
 -/
 
+/-!  ### The `TietzeExtension` class -/
+
+section TietzeExtensionClass
+
+universe u₁ u₂ v w
+
+-- TODO: define *absolute retracts* and then prove they satisfy Tietze extension.
+-- Then make instances of that instead and remove this class.
+/-- A class encoding the concept that a space satisfies the Tietze extension property. -/
+class TietzeExtension (Y : Type v) [TopologicalSpace Y] : Prop where
```

Comment:
This condition follows from the special case, where `X₁` is a closed set in `X₂`, saving us a universe parameter. I'm not sure if this property for `X₁ X₂ : Type v` implies it for any universe.

BTW, is there a natural version of this typeclass that implies `exists_extension_forall_mem_of_closedEmbedding` below? E.g., can we prove that a nonempty convex set in a finite dimensional space has the `TietzeExtension` property?

Reply:
I've implemented the universe parameter reduction.

As for nonempty convex sets in a finite dimensional space, the answer should be *yes*, even with the version I've implemented. The key thing you would need to show is that the nonempty convex set is a retract of the full space. Can we leave that for a future PR, possibly with a TODO?

Reply:
Can we reformulate versions of Tietze Extension theorem like "exists an extension of bounded norm" as "the closed ball is a Tietze extension space"? And same for an `OrdConnected` subset of `Real`.

Reply:
@urkud, yes and no.

Yes: you can prove `TietzeExtension (Metric.closedBall (0 : E) r)` but ...

No: only for `0 ≤ r` and so it wouldn't fire as an instance.

I think the solution, as I've worked out in the latest commit, is to prove the theorem above, and then use it in the proof of `BoundedContinuousFunction.exists_norm_eq_restrict_eq`. Let me know if this is the generality you're looking for.

The same goes for `OrdConnected` (you need the set to be nonempty, so it won't trigger as an instance), but since this applies only to sets in `ℝ`, I think it's fine to leave these more specialized results on their own as is.